update user guide for cpp-api & disable compiling nodejs (#4515)
apache/iotdb: docs/UserGuide/API/Programming-Cpp-Native-API.md

Debian/Ubuntu:

```shell
sudo apt-get install gcc-9 g++-9 libstdc++-9-dev bison flex libboost-all-dev libssl-dev zlib1g-dev npm nodejs
```

CentOS:  

> In CentOS 7.x, the version is to low use yum install gcc,boost and npm, you should compile or download binary package. 

Reviewer: ```suggestion
In CentOS 7.x, the version of gcc, boost and npm installed by yum is too low, therefore you should compile or download these binary packages by yourself. 
```
